samples/apps/copilot-chat-app/webapp/src/libs/semantic-kernel/skMultiUserChat.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft. All rights reserved.
+import * as signalR from "@microsoft/signalr";
+
+export interface UserAsk {
+	input: string;
+	variables: KeyValuePair<string, string>[];
+}
+
+export interface UserAskResult {
+	value: string;
+	variables?: KeyValuePair<string, string>[];
+}
+
+export interface KeyValuePair<K, V> {
+	key: K;
+	value: V;
+}
+
+class SKMultiUserChatConnector {
+	public hubConnection!: signalR.HubConnection;
+	public events!: (
+        onReceiveConversationMessageFE: (userAsk: UserAsk) => void,
+        onReceiveChatSkillAskResultFE: (userAskResult: UserAskResult) => void
+    ) => void;
+	static instance: SKMultiUserChatConnector;
+
+	constructor() {
+		this.setupSignalRConnectionToChatHub();
+        this.registerCommonSignalReconnectionEvents();
+		this.startSignalRConnection(this.hubConnection);
+
+        this.events = ( onReceiveConversationMessageFE, onReceiveChatSkillAskResultFE ) => {
+            this.hubConnection.on("receiveConversationMessageFE", (userAsk) => {
+                onReceiveConversationMessageFE(userAsk);
+            });
+            this.hubConnection.on("receiveChatSkillAskResultFE", (userAskResult) => {
+                onReceiveChatSkillAskResultFE(userAskResult);
+            });
+        };
+	}
+
+	// Set up a SignalR connection to the specified hub URL
+	setupSignalRConnectionToChatHub() {
+		const serviceUrl = process.env.REACT_APP_BACKEND_URI as string;
+		const connectionHubUrl = (new URL('/chatHub', serviceUrl)).toString();
+
+		const signalRConnectionOptions = {
+			skipNegotiation: true,
```

Comment:
Moving conversation here @Vivihung , @adrianwyatt, @glahaye 

I can't establish a connection without skipping negotiations with the server. This is being blocked by the CORS policy of the app. I am not familiar enough with CORS to know if that has some unintended consequences?

Error:
Ensure CORS requests include credentials only when allowed
A cross-origin resource sharing (CORS) request was blocked because it was configured to include credentials but the Access-Control-Allow-Credentials response header of the request or the associated preflight request was not set to true.

Reply:
Changing CORS would look something like this:

    app.UseCors(x => x
        .AllowAnyMethod()
        .AllowAnyHeader()
        .SetIsOriginAllowed(origin => true) // allow any origin
        .AllowCredentials()); // allow credentials
        
But i am not sure what the fallout of that is?

Source: https://www.abrahamberg.com/blog/aspnet-signalr-and-react/ , scroll to Dealing with CORS section